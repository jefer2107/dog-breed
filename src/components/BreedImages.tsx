import { useState } from "react"
import { BreedProps } from "../types"
import { ModalImages } from "./ModalImages"

const BreedImages = ({images, title}: BreedProps) => {
    const [imageSelected, setImageSelected] = useState<string>("")

    return <>
    <h3 className="text-center py-4"> {title} </h3>
    <div className="row container mx-auto">
        {images?.map((x, i) => {
            return(
                <div key={i} className="card col-sm-12 col-md-6 col-lg-4 mx-auto">
                    <div className="image" id="image" onClick={() => setImageSelected(x)} data-bs-toggle="modal" data-bs-target="#exampleModal">
                        <img src={x} className="card-img-top m-1" alt={title} />
                    </div>
                </div>
            )
        })}
    </div>
    <ModalImages imageSelected={imageSelected} />
    </>
}

export default BreedImages