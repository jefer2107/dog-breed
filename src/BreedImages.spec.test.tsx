import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import BreedImages from './components/BreedImages';

const props = {
  title: "title1",
  images: ["images1.jpg", "images2.jpg"]
}

describe('<BreedImages />', () => {
  it('should render images and title correctly', () => {
    
    render(<BreedImages  {...props}/>);

    expect(screen.getByText('title1')).toBeInTheDocument()
    expect(props.images).toEqual(expect.arrayContaining(["images1.jpg","images2.jpg"]))

  });

  it('should send image url correctly when click in image', () => {
    const { container } = render(<BreedImages  {...props}/>)

    expect(container).toMatchSnapshot()

    // const selectImage = jest.fn()
  })
})

