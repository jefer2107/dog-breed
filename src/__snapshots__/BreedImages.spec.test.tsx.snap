// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreedImages /> should send image url correctly when click in image 1`] = `
<div>
  <h3
    class="text-center py-4"
  >
     
    title1
     
  </h3>
  <div
    class="row container mx-auto"
  >
    <div
      class="card col-sm-12 col-md-6 col-lg-4 mx-auto"
    >
      <div
        class="image"
        data-bs-target="#exampleModal"
        data-bs-toggle="modal"
      >
        <img
          alt="title1"
          class="card-img-top m-1"
          src="images1.jpg"
        />
      </div>
    </div>
    <div
      class="card col-sm-12 col-md-6 col-lg-4 mx-auto"
    >
      <div
        class="image"
        data-bs-target="#exampleModal"
        data-bs-toggle="modal"
      >
        <img
          alt="title1"
          class="card-img-top m-1"
          src="images2.jpg"
        />
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalLabel"
    class="modal fade"
    id="exampleModal"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          <img
            class="card-img-top"
            src=""
          />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
